pkg/virt-api/webhooks/mutating-webhook/mutators/clone-create-mutator.go

```diff
@@ -62,16 +63,22 @@ func (mutator *CloneCreateMutator) Mutate(ar *admissionv1.AdmissionReview) *admi
 		return resp
 	}
 
-	raw := ar.Request.Object.Raw
-	vmClone := &clonev1alpha1.VirtualMachineClone{}
+	vmCloneOrig := &clonev1alpha1.VirtualMachineClone{}
 
-	err := json.Unmarshal(raw, &vmClone)
-	if err != nil {
+	if err := json.Unmarshal(ar.Request.Object.Raw, &vmCloneOrig); err != nil {
 		return webhookutils.ToAdmissionResponseError(err)
 	}
 
+	vmClone := vmCloneOrig.DeepCopy()
+
 	mutateClone(vmClone, mutator.targetSuffix)
 
+	if reflect.DeepEqual(vmClone, vmCloneOrig) {
```

Comment:
I'm a little bit hesitant to both using the reflect package (added dependency) and using `DeepEqual` as it usually comes with a non-negligible performance impact.

Either way, I don't think you need to compare the whole object, it's sufficient to compare the 2 Specs as the patch only replace the Spec section.

@xpivarc can you take a look as well? It'd be interesting to see if we're getting a speed-up from this or the opposite.

Reply:
... And one thing that I overlooked: if we decide to go with `DeepEqual` it should rather be `equality.Semantic.DeepEqual`. The implementation in `reflect` may not produce expected results.

Reply:
In future (hopefully this year), we should have a tools to scale test this. I would wait if there is no immediate benefit

Reply:
Replaced the call to `reflect.DeepCopy` with a function to detect whether the target was mutated.